src/models/event-timeline-set.js

```diff
@@ -413,26 +413,31 @@ EventTimelineSet.prototype.addEventsToTimeline = function(events, toStartOfTimel
         const existingIsLive = existingTimeline === this._liveTimeline;
         const timelineIsLive = timeline === this._liveTimeline;
 
-        if (direction === EventTimeline.BACKWARDS && existingIsLive) {
-            // The live timeline should never be spliced into a non-live position.
-            console.warn(
-                "Refusing to set a preceding existingTimeLine on our " +
-                "timeline as the existingTimeLine is live (" + existingTimeline + ")",
-            );
-        } else {
-            timeline.setNeighbouringTimeline(existingTimeline, direction);
-        }
+        const backwardsIsLive = direction === EventTimeline.BACKWARDS && existingIsLive;
+        const forwardsIsLive = direction === EventTimeline.FORWARDS && timelineIsLive;
 
-        if (inverseDirection === EventTimeline.BACKWARDS && timelineIsLive) {
+        if (backwardsIsLive || forwardsIsLive) {
             // The live timeline should never be spliced into a non-live position.
-            console.warn(
-                "Refusing to set our preceding timeline on a existingTimeLine " +
-                "as our timeline is live (" + timeline + ")",
-            );
-        } else {
-            existingTimeline.setNeighbouringTimeline(timeline, inverseDirection);
+            // We use independent logging to better discover the problem at a glance.
+            console.warn({backwardsIsLive, forwardsIsLive}); // debugging
```

Comment:
is this intended to stay in? Should be clear from seeing either of the below messages already?

Reply:
It's intended to stay in to make it slightly easier to troubleshoot from the logs without having to go hunting for messages in the code.